src/ng/animate.js

```diff
@@ -1,6 +1,117 @@
 'use strict';
 
 var $animateMinErr = minErr('$animate');
+var ELEMENT_NODE = 1;
+
+function mergeClasses(a,b) {
+  if (!a && !b) return '';
+  if (!a) return b;
+  if (!b) return a;
+  if (isArray(a)) a = a.join(' ');
+  if (isArray(b)) b = b.join(' ');
+  return a + ' ' + b;
+}
+
+function extractElementNode(element) {
+  for (var i = 0; i < element.length; i++) {
+    var elm = element[i];
+    if (elm.nodeType === ELEMENT_NODE) {
+      return elm;
+    }
+  }
+}
+
+var $$AnimateClassResolverProvider = [function() {
+  var ADD_CLASS = 1;
+  var REMOVE_CLASS = -1;
+
+  this.$get = [function() {
```

Comment:
Functions have a `length` property that basically returns the number of arguments.
The injector tests `fn.length` and does its magic (involving `toString()`) **only** if the returned value is > 0 (aka only if the functionhas arguments).

<sub>Angular, you so smart ! Even core team members can't believe how smart you are :)</sub>